standard/PortalStatistics.lua

```diff
@@ -0,0 +1,1137 @@
+			return year, Table.map(config.customInputs, function(_, mode)
+					return mode, 0
+				end)
+		end)
+
+	local processData = function(item)
+		local year = tonumber(item.date:sub(1, 4))
+		if String.isNotEmpty(item[config.variable]) then
+			local arg = item[config.variable]
+			if earningsData[year][arg] then
+				earningsData[year][arg] = config.processFunction(earningsData[year][arg], item, config)
+			end
+		end
+	end
+
+	local queryParameters = {
+		conditions = conditions:toString(),
+		limit = 5000,
+		query = 'opponenttype, prizemoney, individualprizemoney, date, game',
+	}
+
+	Lpdb.executeMassQuery('placement', queryParameters, processData)
+
+	return Array.map(Array.extractValues(earningsData, Table.iter.spairs), function(value)
+			return Array.map(config.customInputs, function(key)
+						return value[key]
+					end)
+			end)
+end
+
+
+function StatisticsPortal._cacheOpponentPlacementData(args)
+
+	local conditions = ConditionTree(BooleanOperator.all)
+		:add{ConditionNode(ColumnName('liquipediatiertype'), Comparator.neq, 'Qualifier')}
+		:add{ConditionNode(ColumnName('prizemoney'), Comparator.gt, 0)}
+
+	if String.isNotEmpty(args.year) then
+		conditions:add{
+			ConditionNode(ColumnName('date_year'), Comparator.eq, args.year)
+		}
+	end
+
+	local placementConditions = ConditionTree(BooleanOperator.any)
+	for _, allowedPlacement in pairs(args.allowedPlacements) do
+		placementConditions:add{ConditionNode(ColumnName('placement'), Comparator.eq, allowedPlacement)}
+	end
+
+	conditions:add{placementConditions}
+	mw.logObject(conditions:toString())
+	local data = {}
+
+	local queryParameters = {
+		query = 'pagename, shortname, icon, icondark, '
+				.. 'liquipediatier, liquipediatiertype, placement, '
+				.. 'opponentplayers, opponentname, opponenttype',
+		conditions = conditions:toString(),
+		limit = 1000,
+	}
+
+	local function returnOpponent(item)
+		local opponentNames = {}
+		if args.opponentType == Opponent.solo then
+			for playerIndex = 1,10 do
+				local name = string.gsub(item.opponentplayers['p' .. playerIndex] or '', ' ', '_')
+				table.insert(opponentNames, name)
+			end
+		elseif args.opponentType == Opponent.team and item.opponenttype == Opponent.team then
+			local name = string.gsub(item.opponentname or '', ' ', '_')
+			table.insert(opponentNames, name)
+		end
+		return opponentNames
+	end
+
+	local processData = function(item)
+		local placement = string.sub(item.placement, 1, 1)
+		for _, opponent in pairs(returnOpponent(item) or {}) do
+			if not data[opponent] then
+				data[opponent] = {['1'] = 0, ['2'] =  0, ['3'] = 0, ['showWins'] = 0, ['sWinData'] = {}}
+			end
+			if placement == '1' and item.liquipediatier == '1' and item.liquipediatiertype ~= 'Showmatch' then
+				table.insert(data[opponent]['sWinData'], {
+						['icon'] = item.icon,
+						['icondark'] = item.icondark,
+						['pagename'] = item.pagename,
+						['shortname'] = item.shortname
```

Comment:
```suggestion
				table.insert(data[opponent].sWinData, {
						icon = item.icon,
						icondark = item.icondark,
						pagename = item.pagename,
						shortname = item.shortname
```

also check if the all lower capitalization is needed here or if it could be changed to e.g. `iconDark`